gusto/diagnostics.py

```diff
@@ -53,7 +55,10 @@ def l2(f):
 
     @staticmethod
     def total(f):
-        return assemble(f * dx)
+        if len(f.ufl_shape) == 0:
+            return assemble(f * dx)
+        else:
+            pass
```

Comment:
I would definitely like to do something else with this diagnostic, but we can't do that because when the diagnostics are calculated they are calculated for every field (see line 323 of state.py) - so `total` will be passed a vector field at some point. I would rather just not call `total` on vector fields but I couldn't decide on the nicest way to do this - note it also affects the creation of variables in the netcdf file.